src/common/SmartNumericInput/SmartNumericInput.tsx

```diff
@@ -15,7 +15,7 @@ export interface SmartNumericInputProps {
     hidden?: boolean;
 }
 
-export class SmartNumericInput extends React.Component<SmartNumericInputProps, {}> {
+export class SmartNumericInput<T extends SmartNumericInputProps> extends React.Component<T, {}> {
```

Comment:
Why does this component now have a type parameter?

Reply:
In order to add additional props and get use of the methods by extending it.